tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -423,7 +423,12 @@ impl<'writer> Writer<'writer> {
     // We may not want to do that if we choose to expose specialized
     // constructors instead (e.g. `from_string` that stores whether the string
     // is empty...?)
-    pub(crate) fn new(writer: &'writer mut impl fmt::Write) -> Self {
+    //(@kaifastromai) I suppose having dedicated constructors may have certain benefits
+    // but I am not privy to the larger direction of tracing/subscriber.
```

Comment:
If we're committing to making this a public API, I think we can just remove the previous TODO comment.

Reply:
The motivation for not making this public is that I thought it could be valuable for the `Writer` type to be able to indicate whether it was empty or not. This could be used when writing to it, in order to determine whether to emit whitespace before writing more text to the writer. If we were going to track this, however, we couldn't just take an arbitrary `impl fmt::Write`, because there is no way to determine whether that value is empty or not. If we had an explicit `Writer::from_string` constructor, instead, we could check if the string is empty first. Therefore, my preference would be to do that, and keep the `new` function that constructs a `Writer` from an arbitrary value private.

However, I honestly don't care anymore, and I'm happy to approve this PR regardless. If we decide that tracking whether the writer is empty is important, we could make a breaking change in the future.